src/AutoRest.CSharp.V3/AutoRest/Plugins/CSharpGen.cs

```diff
@@ -27,6 +28,7 @@ internal class CSharpGen : IPlugin
     {
         public async Task<bool> Execute(IPluginCommunication autoRest, CodeModel codeModel, Configuration configuration)
         {
+            Directory.CreateDirectory(configuration.OutputFolder);
```

Comment:
I don't like going around autorest in this way.

Reply:
We are not writing files directly same should be with directories.

Reply:
`GeneratedCodeWorkspace.Create` runs `Directory.GetFiles`. So, it really don't make a difference to use some other mechanism instead of `Directory.Create` here.